app/components/Cell/Cell.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Cell({ state, cellIndex, handleClickOnBoard, }) {
+  const { x, y, } = cellIndex;
+  const {
+    colorA,
+    colorB,
+    colorC,
+    cellSideLength,
+    cell,
+    pathStartingPoint,
+    pathEndingPoint,
+  } = state;
+  function getCellColor(currentCell) {
+    if (currentCell.numberOfStepInPath >= 0) {
+      return colorC;
+    }
+    if (!currentCell.isAllowedToBeSteppedOn) {
+      return colorB;
+    }
+    return colorA;
+  }
+
+  const cellStyle = {
+    backgroundColor: getCellColor(cell),
+  };
+  const cellProps = {
+    className: 'cell',
+    style: cellStyle,
+    onClick: e => handleClickOnBoard(x, y, e),
```

Comment:
I prefer this syntax, but it's a matter of taste
```js
onClick(e){
  handleClickOnBoard(x, y, e)
},
```

Reply:
trying it